components/ai_chat/core/common/mojom/ai_chat.mojom

```diff
@@ -746,3 +745,81 @@ interface IAPSubscription {
   GetPurchaseTokenOrderId() => (string token, string order_id);
   SetLinkStatus(int32 status);
 };
+
+// Interfaces for communication between the untrusted content frame and both
+// the Browser and the parent trusted UI frame.
+
+// Trusted WebUI-side handler for messages from the untrusted child frame
+interface ParentUIFrame {
```

Comment:
why combine these? We shouldn't put everything in one mojom file any more than we should put all the headers in one file. They should be separated logically by function and this seems like it made sense in its own file?

Reply:
I think because otherwise there'd be a circular reference for `UntrustedConversationHandler`.

Could we move the structs that are shared between `UntrustedConversationHandler` and the other stuff in here into its own `common.mojom` file and move `UntrustedConversationHandler` into `untrusted_frame.mojom`?

Reply:
PR now rebased and doesn't include any of these changes